mergify_cli/ci_issues/custom_types.py

```diff
@@ -0,0 +1,35 @@
+import re
+import typing
+
+import pydantic
+
+
+def validate_repository_full_name(repository: str) -> str:
+    match = re.match(
+        r"(https?://[\w.-]+/)?(?P<full_name>[\w.-]+/[\w.-]+)/?$",
+        repository,
+    )
+    if match is not None:
+        return match.group("full_name")
+
+    message = "Provide repository URL or 'owner/repo' format"
+    raise ValueError(message)
+
+
+RepositoryT = typing.Annotated[
+    str,
+    pydantic.Field(min_length=1),
+    pydantic.AfterValidator(
+        validate_repository_full_name,
```

Comment:
That does not make much sense, I'm not sure why you'd need an URL for this?

Reply:
I did it because as a user of the CircleCI Orb if you want to pass the repo name dynamically in your workflow you have it exposed only in form of a repository URL through `<<pipeline.trigger_parameters.github_app.repo_url>>`.

Reply:
Right, but then that should be a CircleCI specific-code that does that for CircleCI, not something that *validates* a `Repository` type.

Reply:
Ok I'll let that be handled there then!